crates/oxc_linter/src/rules/typescript/triple_slash_reference.rs

```diff
@@ -0,0 +1,327 @@
+        // We are trying to get the first statement start potioin, falling back to the program end if statement does not exist
+        let comments_range_end = program.body.first().map_or(program.span.end, |v| v.span().start);
+        let comments = ctx.semantic().trivias().comments();
+        let mut refs_for_import = HashMap::new();
+
+        for (start, comment) in comments.range(0..comments_range_end) {
+            let raw = &ctx.semantic().source_text()[*start as usize..comment.end() as usize];
+            if let Some((group1, group2)) = get_attr_key_and_value(raw) {
+                if (group1 == "types" && self.types == TypesOption::Never)
+                    || (group1 == "path" && self.path == PathOption::Never)
+                    || (group1 == "lib" && self.lib == LibOption::Never)
+                {
+                    ctx.diagnostic(TripleSlashReferenceDiagnostic(
+                        group2.to_string(),
+                        Span { start: *start - 2, end: comment.end() },
+                    ));
+                }
+
+                if group1 == "types" && self.types == TypesOption::PreferImport {
+                    refs_for_import.insert(group2, Span { start: *start - 2, end: comment.end() });
+                }
+            }
+        }
+
+        if !refs_for_import.is_empty() {
+            for stmt in &program.body {
+                match stmt {
+                    Statement::Declaration(Declaration::TSImportEqualsDeclaration(decl)) => {
+                        match *decl.module_reference {
+                            TSModuleReference::ExternalModuleReference(ref mod_ref) => {
+                                if let Some(v) =
+                                    refs_for_import.get(mod_ref.expression.value.as_str())
+                                {
+                                    ctx.diagnostic(TripleSlashReferenceDiagnostic(
+                                        mod_ref.expression.value.to_string(),
+                                        *v,
+                                    ));
+                                }
+                            }
+                            TSModuleReference::TypeName(_) => {}
+                        }
+                    }
+                    Statement::ModuleDeclaration(st) => {
+                        if let ModuleDeclaration::ImportDeclaration(ref decl) = **st {
+                            if let Some(v) = refs_for_import.get(decl.source.value.as_str()) {
+                                ctx.diagnostic(TripleSlashReferenceDiagnostic(
+                                    decl.source.value.to_string(),
+                                    *v,
+                                ));
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+}
+
+fn get_attr_key_and_value(raw: &str) -> Option<(String, String)> {
+    if !raw.starts_with('/') {
```

Comment:
```suggestion
    if !raw.starts_with('///') {
```
I think we can check if the beginning of raw contains `///`.

Reply:
all single line comment from trivias will be without `//`,
 so `/// <reference... />` will be `/ <reference... />`

Reply:
I got it!